ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/scan/view/ScanView.kt

```diff
@@ -276,18 +286,27 @@ fun ScanFrame(frameSize: Int) {
 
 @Composable
 fun ScanCameraView(
-    onBack: () -> Unit,
-    cameraProviderFlow: Flow<ProcessCameraProvider>,
-    lifecycleOwner: LifecycleOwner,
-    snackbarHostState: SnackbarHostState
+    onScanned: (result: String) -> Unit,
+    setScanState: (ScanState) -> Unit
 ) {
     val context = LocalContext.current
-    val scope = rememberCoroutineScope()
+    val lifecycleOwner = LocalLifecycleOwner.current
+    val executor = ContextCompat.getMainExecutor(context)
+
+    val cameraProvider = remember { mutableStateOf<ProcessCameraProvider?>(null) }
 
-    val cameraProvider = cameraProviderFlow.collectAsState(initial = null).value
-    if (null == cameraProvider) {
+    val cameraProviderFuture = remember { ProcessCameraProvider.getInstance(context) }
```

Comment:
Here I had to switch from the Flow way back to Future way of implementation, as it didn't get triggered after changing the permission state from outside of the app - Settings app.

Reply:
Good to know.  I don't love the use of a future callback that's not tied to our Compose scope.

Put another way, the views can be torn down but the listenable future might still trigger a callback here.  Although the one line assignment is probably safe, the entire lambda is risky if another developer comes along and decides to touch something else inside of there.  (Similar to how AsyncTask or lambdas in Activity or Fragment that do something ignoring the lifecycle can hit a NullPointerException or other exceptions, if the Activity/Fragment finishes before the lambda is invoked and the lambda doesn't check for that).

I'm going to poke at this a little more to see if we can come up with something

Reply:
Yes, understand the problem. Check it out, if you find a better way of doing this. Thank you very much!